File: factorial/php/TestFactorial.php
<?php
declare(strict_types=1);
use PHPUnit\Framework\TestCase;

require('factorial.php');

final class TestFactorial extends TestCase
{
    public function testReturnsExpectedResponsesWhenInputIsInvalid():void
    {
        $this->assertEquals('Invalid input!', factorial());
        $this->assertEquals('Invalid input!', factorial(-1));
        $this->assertEquals('Invalid input!', factorial(4.8));
        $this->assertEquals('Invalid input!', factorial('abd'));
    }

    public function testReturnsExpectedResponsesWhenInputIsValid():void
    {
        $this->assertEquals(1, factorial(1));
        $this->assertEquals(1, factorial(1));
        $this->assertEquals(2, factorial(2));
        $this->assertEquals(6, factorial(3));
        $this->assertEquals(24, factorial(4));
        $this->assertEquals(3628800, factorial(10));
    }
}

File: programming/factorial/python/test_factorial.py
from factorial import factorial


def test_factorial():
    assert factorial(-1) == 'Invalid input!'
    assert factorial('a') == 'Invalid input!'
    assert factorial(-3) == 'Invalid input!'
    assert factorial(2.9) == 'Invalid input!'
    assert factorial(0) == 1
    assert factorial(1) == 1
    assert factorial(2) == 2
    assert factorial(4) == 24
    assert factorial(10) == 3628800
